### ISSUE TITLE
Document dependency handling in remote extensions

### ISSUE THREAD

**Issue by connor4312** [microsoft/vscode-docs#3417]
https://github.com/microsoft/vscode/issues/91109#issuecomment-589392164

**Comment by alexdima** [comment]
Here is an explanation:
* extensions can export API from their `activate` function.
* this API will become available to all extensions running in the same extension host.
* normally, consumer extensions would declare in their `package.json` that they depend on the providing extension using the `extensionDependencies` property.

When dealing with remote scenarios, it is possible that an extension running remotely has an extension dependency on an extension running locally. For example, the local extension exposes a command that is critical to the good functioning of the remote extension. In this case, we recommend that the remote extension declares the local extension as an `extensionDependency`, but the problem is that the extensions run on two different extension hosts, which means that the API from the provider is not available to the consumer. We therefore require that the providing extension **gives up entirely the ability to export API using `"api": "none"`** in their `package.json`. The extensions will communicate using vscode commands (which are async).

---

So, in other words, an extension that uses `"api": "none"` gives up the ability to return API from its `activate` method. By doing this, extensions that execute on other extension hosts can take dependency on them and will be activated.
